pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -114,6 +114,23 @@ func (service *SubnetPortService) loadNSXMacPool() error {
 	return nil
 }
 
+func (service *SubnetPortService) portAlreadyRealized(obj interface{}) bool {
+	switch o := obj.(type) {
+	case *v1alpha1.SubnetPort:
+		if len(o.Status.Conditions) > 0 && o.Status.Conditions[0].Reason == "SubnetPortReady" && o.Status.Conditions[0].Status == v1.ConditionTrue && len(o.Status.Attachment.ID) > 0 {
```

Comment:
Would it be better to use loop in case we may add more conditions to SubnetPort in the future?

Reply:
Thanks, done